src/__tests__/lib/exampleData.js

```diff
@@ -263,6 +264,28 @@ export const makeSubscription = (args: {| stream?: Stream |} = Object.freeze({})
 /** A subscription to eg.stream. */
 export const subscription: Subscription = makeSubscription();
 
+/** Construct mock UnreadStreamsState. */
+export const makeUnreadStreamsState = (
+  mockData: {|
+    streamId: number,
+    topics: {|
+      topicName: string,
+      messageIds: number[],
+    |}[],
+  |}[],
+): UnreadStreamsState =>
+  Immutable.Map(
+    mockData.map(unreadStream => [
+      unreadStream.streamId,
+      Immutable.Map(
+        unreadStream.topics.map(unreadTopic => [
+          unreadTopic.topicName,
+          Immutable.List(unreadTopic.messageIds),
+        ]),
+      ),
+    ]),
+  );
```

Comment:
Instead of directly building up the internal data structure that `state.unread.streams` comes in, let's use the unreads reducer to build it up. That way:
* The details of the data structure stay internal to the unreads model. This makes it easier to change that data structure around later (as we recently did, in fact, earlier this year.)
* By keeping those details out, it also keeps this code that's building the data more abstract in a way that I think will make it more readable: it'll say "start from an initial/empty state, then insert each of the messages in this list".
* Because the reducer takes basically message objects, I think this will ultimately help simplify the tests that use this, too: instead of referring to a stream by an ID here and a name there, and counting on tying those together elsewhere in the test, or using the same ID or name in several places and implicitly using that they're the same value, they can use a real `Stream` object throughout.

Reply:
For an example, see:
https://github.com/zulip/zulip-mobile/blob/31c4bc808053532499399e0dc241810e4a0113b9/src/unread/__tests__/unreadModel-test.js#L149-L160

For another example:
https://github.com/zulip/zulip-mobile/blob/31c4bc808053532499399e0dc241810e4a0113b9/src/unread/__tests__/unread-testlib.js#L36-L58

Reply:
> Instead of directly building up the internal data structure that `state.unread.streams` comes in, let's use the unreads reducer to build it up.

(This isn't a principle we have written down anywhere -- in fact I really only started doing it in our code just recently, with that refactor to this very unreads model. I think it becomes more important as our data structures get interesting, like that one is. @chrisbobbe noticed it at the time and we had a conversation about it, which might be informative to read and I should perhaps turn into a style-guide entry or something; maybe it was on one of the PRs where I was making those changes?)